go/runtime/txpool/txpool.go

```diff
@@ -894,6 +930,7 @@ func New(
 		host:                 host,
 		history:              history,
 		txPublisher:          txPublisher,
+		consensus:            consensus,
```

Comment:
Feels like the pool shouldn't have this dependency. Maybe the node should start the pool in the `worker` method once consensus is synced, not sure.

Reply:
This is the only way to enable calling the `consensus.Synced` method.  I'm pretty sure the node can't move txpool init to after the consensus is synced because the txpool methods can be called before that happens and if it's not inited, they would just crash.

Reply:
Agree. I was just thinking about how would you unit-test this. And you would need to mock the consensus. On the other hand, starting the worker from the outside once consensus is synced, is cleaner.

Reply:
This is a good point, the `TxPool` is started from the common worker (`worker/common/committee/node.go`) in its `Start` method. You could instead start it after the common worker checks that consensus is synced.